### PR TITLE
Update README, Module and package .md files

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$title 

Fixes ballerina-platform/ballerina-standard-library#1710
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   [![codecov](https://codecov.io/gh/ballerina-platform/module-ballerinax-oracledb/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerinax-oracledb)
 
-The OracleDB library is one of the standard library packages of the<a target="_blank" href="https://ballerina.io/"> Ballerina</a> language.
+The OracleDB library is one of the external library packages of the <a target="_blank" href="https://ballerina.io/"> Ballerina</a> language.
 
-This provides the functionality required to access and manipulate data stored in an OracleDB database.  
+This provides the functionality required to access and manipulate data stored in an Oracle database.
 
-For more information on the operations supported by the `oracledb:Client`, which includes the below, go to the [OracleDB Package](https://ballerina.io/learn/api-docs/ballerina/oracledb/).
+For more information on the operations supported by the `oracledb:Client`, which include the below, go to [The OracleDB Package](https://docs.central.ballerina.io/ballerinax/oracledb/latest).
```
Comment (praneesha) [inline README.md:14]:
```suggestion
For more information on the operations supported by the `oracledb:Client`, which include the below, go to [The `oracledb` package](https://docs.central.ballerina.io/ballerinax/oracledb/latest).
```

File: README.md [hunk 2]
```diff
@@ -22,24 +22,24 @@ For more information on the operations supported by the `oracledb:Client`, which
...
+Issues and Projects tabs are disabled for this repository as this is part of the Ballerina Standard Library. To report bugs, request new features, start new discussions, view project boards, etc. please visit Ballerina Standard Library [parent repository](https://github.com/ballerina-platform/ballerina-standard-library).
 
 This repository only contains the source code for the package.
 
 ## Building from the Source
 
 ### Setting Up the Prerequisites
 
-1. Download and install the Java SE Development Kit (JDK) version 11 (from one of the following locations).
+1. Download and install Java SE Development Kit (JDK) version 11 (from one of the following locations).
```
Comment (praneesha) [inline README.md:35]:
```suggestion
1. Download and install the Java SE Development Kit (JDK) version 11 (from one of the following locations).
```

File: README.md [hunk 3]
```diff
@@ -22,24 +22,24 @@ For more information on the operations supported by the `oracledb:Client`, which
...
    * [Oracle](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html)
    * [OpenJDK](http://openjdk.java.net/install/index.html)
 
-2. Download and install [Docker](https://www.docker.com/get-started).
-   
-3. Export your Github Personal access token with the read package permissions as follows.
-        
+2. Download and install [Docker](https://www.docker.com/get-started)
```
Comment (praneesha) [inline README.md:39]:
```suggestion
2. Download and install [Docker](https://www.docker.com/get-started).
```

File: README.md [hunk 4]
```diff
@@ -22,24 +22,24 @@ For more information on the operations supported by the `oracledb:Client`, which
...
+
+3. Export Github Personal access token with read package permissions as follows,
```
Comment (praneesha) [inline README.md:41]:
```suggestion
3. Export your Github Personal access token with the read package permissions as follows.
```

File: README.md [hunk 5]
```diff
@@ -68,24 +68,44 @@ Execute the commands below to build from the source.
...
+   execute | execute-basic <br> execute-params
+   batch-execute | batch-execute
+   query | query-simple-params<br>query-complex-params
+   procedures | procedures
+   datetime | datetime
+   custom-object | custom-object
+   custom-varray | custom-varray
 
-5. To disable some specific groups during the test:
+5. To disable some specific groups during test,
```
Comment (praneesha) [inline README.md:80]:
```suggestion
5. To disable some specific groups during the test:
```

File: README.md [hunk 6]
```diff
@@ -68,24 +68,44 @@ Execute the commands below to build from the source.
...
 
         ./gradlew clean build -Pdisable-groups=<Comma separated groups/test cases>
 
 6. To debug the tests:
 
         ./gradlew clean build -Pdebug=<port>
+        ./gradlew clean test -Pdebug=<port>
+
+7. To debug the package with Ballerina language:
```
Comment (praneesha) [inline README.md:89]:
```suggestion
7. To debug the package with the Ballerina language:
```

File: README.md [hunk 7]
```diff
@@ -68,24 +68,44 @@ Execute the commands below to build from the source.
...
+
+        ./gradlew clean build -PbalJavaDebug=<port>
+        ./gradlew clean test -PbalJavaDebug=<port>
+
+8. Publish ZIP artifact to the local `.m2` repository:
```
Comment (praneesha) [inline README.md:94]:
```suggestion
8. Publish the ZIP artifact to the local `.m2` repository:
```

File: ballerina/Module.md [hunk 8]
```diff
@@ -0,0 +1,426 @@
+## Overview
+
+This module provides the functionality required to access and manipulate data stored in an Oracle database.
+
+### Prerequisite
+Add the OracleDB thin driver `ojdbc8.jar` along with `xdb.jar` and `xmlparserv2.jar` as native library dependencies in your Ballerina project's `Ballerina.toml` file.
+It is recommended to use an oracle thin driver `ojdbc8.jar` version greater than 12.2.0.1.
+
+Follow one of the following ways to add the JAR in the file:
```
Comment (praneesha) [inline ballerina/Module.md:9]:
```suggestion
Follow one of the ways below to add the JAR in the file:
```

File: ballerina/Module.md [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+
+* Download the JAR and update the path
```
Comment (praneesha) [inline ballerina/Module.md:11]:
```suggestion
* Download the JAR and update the path.
```

File: ballerina/Module.md [hunk 10]
```diff
@@ -0,0 +1,426 @@
...
+    ```
+    [[platform.java11.dependency]]
+    path = "PATH"
+    ```
+
+* Add JAR with a maven dependency params
```
Comment (praneesha) [inline ballerina/Module.md:17]:
```suggestion
* Add JAR with the Maven dependency params.
```

File: ballerina/Module.md [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+    version = "21.1.0.0"
+  
+    [platform.java11.dependency]]
+    groupId = "com.oracle.database.xml"
+    artifactId = "xmlparserv2"
+    version = "12.2.0.1"
+    ```
+
+### Client
+To access a database, you must first create a
```
Comment (praneesha) [inline ballerina/Module.md:36]:
```suggestion
To access a database, you must first create an
```

File: ballerina/Module.md [hunk 12]
```diff
@@ -0,0 +1,426 @@
...
+
+```ballerina
+oracledb:Client|sql:Error dbClient = new (user = "adminUser", password = "adminPassword",
+                              connectionPool = {maxOpenConnections: 5});
+```
+
+#### Connection Pool Handling
+
+All ballerina database modules share the same connection pooling concept and there are three possible scenarios for
+connection pool handling.  For its properties and possible values, see the [`sql:ConnectionPool`](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool).
```
Comment (praneesha) [inline ballerina/Module.md:99]:
```suggestion
connection pool handling. For its properties and possible values, see the [`sql:ConnectionPool`](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool).
```

File: ballerina/Module.md [hunk 13]
```diff
@@ -0,0 +1,426 @@
...
+
+1. Global shareable default connection pool
```
Comment (praneesha) [inline ballerina/Module.md:101]:
```suggestion
1. Global shareable default connection pool
```

File: ballerina/Module.md [hunk 14]
```diff
@@ -0,0 +1,426 @@
...
+
+   If you do not provide the `poolOptions` field when creating the database client, a globally-shareable pool will be
+   created for your database unless a connection pool matching with the properties you provided already exists.
```
Comment (praneesha) [inline ballerina/Module.md:104]:
```suggestion
   created for your database unless a connection pool matching the properties you provided already exists.
```

File: ballerina/Module.md [hunk 15]
```diff
@@ -0,0 +1,426 @@
...
+   an unsharable connection pool will be created.
+
+    ```ballerina
+    oracledb:Client|sql:Error dbClient = new (user = "adminUser", password = "adminPassword",
+                                           connectionPool = { maxOpenConnections: 5 });
+    ```
+
+3. Local, shareable connection pool
+
+   If you create a record of type `sql:ConnectionPool` and reuse that in the configuration of multiple clients,
```
Comment (praneesha) [inline ballerina/Module.md:122]:
```suggestion
   If you create a record of the`sql:ConnectionPool` type and reuse that in the configuration of multiple clients,
```

File: ballerina/Module.md [hunk 16]
```diff
@@ -0,0 +1,426 @@
...
+   for each set of clients that connects to the same database instance with the same set of properties, a shared
```
Comment (praneesha) [inline ballerina/Module.md:123]:
```suggestion
   for each set of clients that connects to the same database instance with the same set of properties, a shared
```

File: ballerina/Module.md [hunk 17]
```diff
@@ -0,0 +1,426 @@
...
+error? e = dbClient.close();
+```
+Or
+```ballerina
+check dbClient.close();
+```
+
+### Database Operations
+
+Once the client is created, database operations can be executed through that client. This module defines the interface
```
Comment (praneesha) [inline ballerina/Module.md:161]:
```suggestion
Once the client is created, the database operations can be executed through that client. This module defines the interface
```

File: ballerina/Module.md [hunk 18]
```diff
@@ -0,0 +1,426 @@
...
+and common properties that are shared among multiple database clients.  It also supports querying, inserting, deleting,
+updating, and batch updating data. 
+
+#### Creating Tables
+
+This sample creates a table with two columns. One column is of type `int` and the other is of type `varchar`.
```
Comment (praneesha) [inline ballerina/Module.md:167]:
```suggestion
This sample creates a table with two columns. One column is of the `int` type and the other is of the `varchar` type.
```

File: ballerina/Module.md [hunk 19]
```diff
@@ -0,0 +1,426 @@
...
+The `CREATE` statement is executed via the `execute` remote function of the client.
+
+```ballerina
+// Create the ‘Students’ table with the  ‘id’, 'name', and ‘age’ fields.
+sql:ExecutionResult result = check dbClient->execute("CREATE TABLE student(id INT AUTO_INCREMENT, " +
+                         "age INT, name VARCHAR(255), PRIMARY KEY (id))");
+//A value of the`sql:ExecutionResult` type is returned for 'result'.
```
Comment (praneesha) [inline ballerina/Module.md:174]:
```suggestion
//A value of the`sql:ExecutionResult` type is returned for the 'result'. 
```

File: ballerina/Module.md [hunk 20]
```diff
@@ -0,0 +1,426 @@
...
+```
+
+#### Querying Data
+
+These samples show how to demonstrate the different usages of the `query` operation and query the
+database table and obtain the results.
+
+This sample demonstrates querying data from a table in a database.
+First, a type is created to represent the returned result set. This record can be defined as an open or a closed record
+according to the requirement. If an open record is defined, the returned stream type will include both defined fields
```
Comment (praneesha) [inline ballerina/Module.md:243]:
```suggestion
according to the requirement. If an open record is defined, the returned stream type will include both the defined fields
```

File: ballerina/Module.md [hunk 21]
```diff
@@ -0,0 +1,426 @@
...
+in the record and additional database columns fetched by the SQL query which are not defined in the record.
```
Comment (praneesha) [inline ballerina/Module.md:244]:
```suggestion
in the record and additional database columns fetched by the SQL query, which are not defined in the record.
```

File: ballerina/Module.md [hunk 22]
```diff
@@ -0,0 +1,426 @@
...
+Note the mapping of the database column to the returned record's property is case-insensitive if it is defined in the
+record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional Column names
```
Comment (praneesha) [inline ballerina/Module.md:246]:
```suggestion
record (i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional column names
```

File: ballerina/Module.md [hunk 23]
```diff
@@ -0,0 +1,426 @@
...
+added to the returned record as in the SQL query. If the record is defined as a close record, only defined fields in the
```
Comment (praneesha) [inline ballerina/Module.md:247]:
```suggestion
are added to the returned record as in the SQL query. If the record is defined as a close record, only the defined fields in the
```

File: ballerina/Module.md [hunk 24]
```diff
@@ -0,0 +1,426 @@
...
+record are returned or gives an error when additional columns present in the SQL query. Next, the `SELECT` query is executed
```
Comment (praneesha) [inline ballerina/Module.md:248]:
```suggestion
record are returned or gives an error when additional columns are present in the SQL query. Next, the `SELECT` query is executed
```

File: ballerina/Module.md [hunk 25]
```diff
@@ -0,0 +1,426 @@
...
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<Student, sql:Error?> resultStream = dbClient->query(query);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(Student student) {
+   //Can perform any operations using 'student' and can access any fields in the returned record of type Student.
```
Comment (praneesha) [inline ballerina/Module.md:272]:
```suggestion
   //Can perform any operations using 'student' and can access any fields in the returned record of type `Student`.
```

File: ballerina/Module.md [hunk 26]
```diff
@@ -0,0 +1,426 @@
...
+string name = "John";
+sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
+sql:ExecutionResult result = check dbClient->execute(query);
+```
+
+#### Batch Updating Data
+
+This sample demonstrates how to insert multiple records with a single `INSERT` statement that is executed via the
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and
+parameterized SQL query as same as the  above `execute` operations.
```
Comment (praneesha) [inline ballerina/Module.md:345]:
```suggestion
a parameterized SQL query as same as the above `execute` operations.
```

File: ballerina/Module.md [hunk 27]
```diff
@@ -0,0 +1,426 @@
...
+```
+Note that you have to explicitly invoke the close operation on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
+
+### OracleDB specific Custom data types
+ 
+#### Interval Types
+
+OracleDB has two `INTERVAL` data types `INTERVAL YEAR TO MONTH` and `INTERVAL DAY TO SECOND` to store the various `INTERVAL` periods. 
+
+ballerina equivalent types as follows
```
Comment (praneesha) [inline ballerina/Module.md:392]:
```suggestion
The equivalent types in Ballerina are as follows.
```

File: ballerina/Module.md [hunk 28]
```diff
@@ -0,0 +1,426 @@
...
+
+#### Creating Tables
+
+This sample creates a table with two columns. One column is of the `int` type and the other is of the `varchar` type.
+The `CREATE` statement is executed via the `execute` remote function of the client.
+
+```ballerina
+// Create the ‘Students’ table with the  ‘id’, 'name', and ‘age’ fields.
+sql:ExecutionResult result = check dbClient->execute("CREATE TABLE student(id INT AUTO_INCREMENT, " +
+                         "age INT, name VARCHAR(255), PRIMARY KEY (id))");
```
Comment (daneshk) [inline ballerina/Module.md:173]: Don't use string queries. use parameterized queries

Comment (daneshk) [reply ballerina/Module.md:173]: check and correct all places

Comment (lahirusamith) [reply ballerina/Module.md]: corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/27/commits/bcf2f27f652970e8e3740ebf430790b44d9c2978

**Review by daneshk — APPROVED** [review]

**Pull request merged**